ibis/backends/duckdb/__init__.py

```diff
@@ -172,7 +192,17 @@ def register(
         ir.Table
             The just-registered table
         """
-        if isinstance(source, (str, Path)):
+        if isinstance(source, str) and source.startswith("s3://"):
```

Comment:
this extra conditional is a bit annoying, but it's necessary since we need a cursor to register the arrow dataset with (and we also need the dataset itself, so if we're keeping the dispatcher "pure" we need to add in a fork here